parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -744,4 +744,80 @@ EOF
     fi
 }
 
+getPrimaryNicIP() {
+    local sleepTime=1
+    local maxRetries=10
+    local i=0
+    local ip=""
+
+    while [[ $i -lt $maxRetries ]]; do
```

Comment:
nit: we already have a helper which does something very similar which you can probably use instead: https://github.com/Azure/AgentBaker/blob/45392616eccd6b2bc9b8b9ed89416a1ba75715c4/parts/linux/cloud-init/artifacts/cse_helpers.sh#L128C1-L140C2

Reply:
Actually I also noticed the `retrycmd_if_failure` function, but it does not support return value via `echo "$ip"`, so I manually added some retry logic here.

One more point I just realized, do you have any suggestion on whether we should add retry for manipulating iptables rule in `ensureIMDSRestrictionRule` and `disableIMDSRestriction`?

Reply:
probably a good question for @wedaly

Reply:
`Actually I also noticed the retrycmd_if_failure function, but it does not support return value via echo "$ip", so I manually added some retry logic here.` fair enough... though you should be able to accomplish this using the existing function if you use a global var, though that is somewhat messy as well

I'm fine either way I suppose